File: cpp/SelectionSort.cpp
#include<bits/stdc++.h> 

using namespace std; 
void SelectionSort(vector<int>&arr){
    int n = arr.size();

    for(int i = 0 ;i < n-1 ; i++){
        int min_idx = i;
        for(int j = i+1 ; j < n ; j++){
            if(arr[j] < arr[min_idx]){
                min_idx = j;
            }
        }

        swap(arr[i],arr[min_idx]);
    }
}

int main(){
cout<<"Enter size of the array : ";

int n;
cin>>n;
vector<int>arr;

for(int i = 0 ; i < n ; i++){
    int u ;
    cin>>u;
    arr.push_back(u);
}

cout<<"Array before Selection Sort : "<<endl;
for(auto it : arr){
    cout<<it<<" ";
}
cout<<endl;

SelectionSort(arr);

cout<<"Array After Selection Sort : "<<endl;
for(auto it : arr){
    cout<<it<<" ";
}
cout<<endl;

return 0 ; 

}

File: java/SelectionSort.java
import java.util.*;

public class SelectionSort {

    // Selection Sort Function
    static void selectionSort(int[] arr) {

        int n = arr.length;

        for (int i = 0; i < n - 1; i++) {

            int minIndex = i;

            // Find minimum element
            for (int j = i + 1; j < n; j++) {

                if (arr[j] < arr[minIndex]) {

                    minIndex = j;
                }
            }

            // Swap elements
            int temp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = temp;
        }
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter size of the array: ");

        int n = sc.nextInt();

        int[] arr = new int[n];

        System.out.println("Enter array elements:");

        for (int i = 0; i < n; i++) {

            arr[i] = sc.nextInt();
        }

        System.out.println(
                "\nArray before Selection Sort:"
        );

        for (int num : arr) {

            System.out.print(num + " ");
        }

        System.out.println();

        // Call Selection Sort
        selectionSort(arr);

        System.out.println(
                "\nArray after Selection Sort:"
        );

        for (int num : arr) {

            System.out.print(num + " ");
        }

        sc.close();
    }
}